hops/featurestore_impl/core.py

```diff
@@ -1966,6 +1966,60 @@ def _do_get_redshift_featuregroup(storage_connector_name, query, featurestore_me
                                                                    .format(storage_connector.type))
 
 
+def _do_add_metadata(featuregroup_name, name, value, featurestore=None, featuregroup_version=1):
+    """
+    Attach custom metadata to a feature group
+
+    Args:
+        :featuregroup_name: the name of the featuregroup
+        :name: the name of the extended metadata
+        :value: the value of the extended metadata
+        :featurestore: the featurestore where the featuregroup resides
+        :featuregroup_version: the version of the featuregroup
+
+    Returns:
+          None
+    """
+    featuregroup_id = _get_featuregroup_id(featurestore, featuregroup_name, featuregroup_version)
+    featurestore_id = _get_featurestore_id(featurestore)
+    rest_rpc._add_metadata(featurestore_id, featuregroup_id, name, value)
```

Comment:
As mentioned in hops-util. I believe it would be better to send the entire dict over to the backend in one rest call to ensure consistency  and also avoid multiple round trips.

Reply:
I agree, however, it will won't ensure the consistency either. Since for each key value pair a file system operation will run to add/get/remote extended attributes.

Reply:
Alright nothing we can do about it right now then.